OpenRA.Mods.Common/Traits/GrantConditionWhenDockClient.cs

```diff
@@ -0,0 +1,82 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public sealed class GrantConditionWhenDockClientInfo : TraitInfo
```

Comment:
This name is a bit weird. Why not `GrantConditionOnDockInfo`?

Reply:
It is sad that we have IDockClient and IDockHost, I tried to combine those but only make the code more confusing and it ends up with splited into two.

Reply:
Fixed by using comment below.